tests/app/main/helpers/test_services.py

```diff
@@ -0,0 +1,110 @@
+import mock
+import pytest
+
+from app.main.helpers.services import copy_service_from_previous_framework
+
+
+class CustomAbortException(Exception):
+    """
+    Custom exception used in tests to mimick `aborts` behaviour of halting code execution. Catching this specific
+    exception in the tests allows us to know an exception is being thrown for the expected reason.
+    """
+    pass
+
+
+class TestCopyServiceFromPreviousFramework():
+    def setup_method(self):
+        self.abort_patch = mock.patch('app.main.helpers.services.abort')
+        self.abort = self.abort_patch.start()
+        self.abort.side_effect = CustomAbortException()
+
+        self.get_supplier_framework_info_patch = mock.patch('app.main.helpers.services.get_supplier_framework_info')
+        self.get_supplier_framework_info = self.get_supplier_framework_info_patch.start()
+        self.get_supplier_framework_info.return_value = True
+
+        self.api_client_mock = mock.Mock()
+        self.content_loader_mock = mock.Mock()
+        self.content_loader_mock.get_metadata.side_effect = [['list', 'of', 'questions'], 'dos-cloud-IX']
+
+    def previous_service(self, **kwargs):
+        service_data = {
+            'id': 4444244,
+            'lotSlug': 'digital-sausages',
+            'frameworkSlug': 'dos-cloud-IX',
+            'copiedToFollowingFramework': False,
+            'supplierId': 1234,
+        }
+        service_data.update(kwargs)
+        return {'services': service_data}
+
+    def assert_404_and_no_copy(self, assertion_error=None):
+        assert self.abort.call_args_list == [mock.call(404)], assertion_error
+        assert self.api_client_mock.copy_draft_service_from_existing_service.call_args_list == []
+
+    @mock.patch('app.main.helpers.services.current_user')
+    def test_correctly_calls_api_client(self, current_user):
+        current_user.name = 'Mr Sausages'
+        current_user.supplier_id = 1234
+
+        self.api_client_mock.get_service.return_value = self.previous_service()
+
+        copy_service_from_previous_framework(
+            self.api_client_mock, self.content_loader_mock, 'dos-cloud-X', 'digital-sausages', 4444244
+        )
+
+        assert self.content_loader_mock.get_metadata.call_args_list == [
```

Comment:
I guess this is technically an implementation detail, so it probably doesn't need asserting. As long as we end up calling the api client with the correct args, it should be fine, I think?

Reply:
Yep, have removed, and the assertation on the api client `get_service` call.